dbt/adapters/bigquery/relation_configs/materialized_view.py

```diff
@@ -0,0 +1,165 @@
+            "hours_to_expiration": model_node.config.extra.get("hours_to_expiration"),
+            "max_staleness": model_node.config.extra.get("max_staleness"),
+            "allow_non_incremental_definition": model_node.config.extra.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": model_node.config.extra.get("kms_key_name"),
+            "friendly_name": model_node.config.extra.get("friendly_name"),
+            "description": model_node.config.extra.get("description"),
+            "labels": model_node.config.extra.get("labels"),
+        }
+
+        autorefresh_value = model_node.config.extra.get("enabled_refresh")
+        if autorefresh_value is not None:
+            if isinstance(autorefresh_value, bool):
+                config_dict["enable_refresh"] = autorefresh_value
+            elif isinstance(autorefresh_value, str):
+                lower_autorefresh_value = autorefresh_value.lower()
+                if lower_autorefresh_value == "true":
+                    config_dict["enable_refresh"] = True
+                elif lower_autorefresh_value == "false":
+                    config_dict["enable_refresh"] = False
+                else:
+                    raise ValueError(
+                        "Invalide enable_refresh representation. Please used excepted value ex.(True, 'true', 'True')"
+                    )
+            else:
+                raise TypeError("Invalid autorefresh value: expecting boolean or str.")
+
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
+        materialized_view: agate.Row = cls._get_first_row(
+            relation_results.get("materialized_view")  # type: ignore[arg-type]
+        )
+
+        config_dict = {
+            "materialized_view_name": materialized_view.get("materialized_view_name"),
+            "schema_name": materialized_view.get("schema"),
+            "database_name": materialized_view.get("database"),
+            "cluster_by": materialized_view.get("cluster_by"),
+            "partition_by": materialized_view.get("partition_by"),
+            "enable_refresh": materialized_view.get("enabled_refresh"),
+            "refresh_interval_minutes": materialized_view.get("refresh_interval_minutes"),
+            "hours_to_expiration": materialized_view.get("hours_to_expiration"),
+            "max_staleness": materialized_view.get("max_staleness"),
+            "allow_non_incremental_definition": materialized_view.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": materialized_view.get("kms_key_name"),
+            "friendly_name": materialized_view.get("friendly_name"),
+            "description": materialized_view.get("description"),
+            "labels": materialized_view.get("labels"),
+        }
+
+        return config_dict
```

Comment:
@mikealfare Do we have give every `Optional` field conditional logic here to check for it?

Reply:
I don't know that I understand the question. Are you concerned that we're creating keys for settings that the user did not set? That's handled in `from_dict` in the base class in `dbt-core`, it filters out any settings where the value is `None`; basically this assumes that `None` is never a setting for any user config, hence it should be replaced with the default.

Reply:
Of course, to clarify a little more I was curious if any of these looked like they might fall into a similar format as `sort` and `dist` in `dbt-redshift` especially the ones like `enable_autorefresh` where the technical guide are wanting to make classes for them.